src/downloadthread.cpp

```diff
@@ -155,23 +160,92 @@ bool DownloadThread::_openAndPrepareDevice()
+                    for (const auto& mountpoint : i.mountpoints)
+                    {
+                        QString driveLetter = QString::fromStdString(mountpoint);
+                        if (driveLetter.endsWith("\\"))
+                            driveLetter.chop(1);
+                        
+                        qDebug() << "Attempting to unmount drive" << driveLetter;
+                        
+                        // Try to lock and unlock the volume to force unmount
+                        WinFile tempFile;
+                        tempFile.setFileName("\\\\.\\" + driveLetter);
+                        if (tempFile.open(QIODevice::ReadWrite))
+                        {
+                            if (tempFile.lockVolume())
+                            {
+                                tempFile.unlockVolume();
+                                qDebug() << "Successfully unlocked volume" << driveLetter;
+                            }
+                            tempFile.close();
+                        }
+                        
+                        // Give the system time to process the unmount
+                        QThread::msleep(500);
+                    }
+                    break;
+                }
+            }
 
-            if (proc.exitCode())
+            // Now run diskpart with retry logic
+            bool diskpartSuccess = false;
+            for (int attempt = 1; attempt <= 3; attempt++)
             {
-                emit error(tr("Error running diskpart: %1").arg(QString(proc.readAllStandardError())));
+                qDebug() << "Running diskpart attempt" << attempt;
+                
+                QProcess proc;
+                proc.start("diskpart", QStringList());
+                if (!proc.waitForStarted(5000))
+                {
+                    qDebug() << "Failed to start diskpart";
+                    continue;
+                }
+                
+                proc.write("select disk "+_nr+"\r\n"
+                                "clean\r\n"
+                                "rescan\r\n");
+                proc.closeWriteChannel();
+                
+                if (!proc.waitForFinished(30000)) // 30 second timeout
+                {
+                    qDebug() << "diskpart timed out";
+                    proc.kill();
+                    continue;
+                }
+
+                if (proc.exitCode() == 0)
+                {
+                    diskpartSuccess = true;
+                    qDebug() << "diskpart succeeded on attempt" << attempt;
+                    break;
+                }
+                else
+                {
+                    QString errorOutput = QString(proc.readAllStandardError());
+                    qDebug() << "diskpart failed on attempt" << attempt << "with error:" << errorOutput;
+                    
+                    // Wait before retry
+                    QThread::msleep(1000 * attempt); // Progressive delay
```

Comment:
As a micro-optimisation I guess you could skip the 3-second sleep when `attempt == 3`, as there aren't going to be any more retries?